### PR TITLE
Check question content

### LINKED ISSUE #108: Question changeset: Add verification according to question type

### PR DESCRIPTION
**Pull request by tuesmiddt**
Fixes #108 

Depends on #143

### CODE REVIEW

**Comment by indocomsoft** [comment]
Lel conflicts everywhere

File: test/cadet/assessments/question_test.exs [hunk 1]
```diff
@@ -1,49 +1,130 @@
...
+      assert({:ok, _} = res, inspect(res, pretty: true))
+    end
+  end
+
+  describe "invalid changesets" do
+    test "missing params", %{
+      valid_mcq_params: mcq_params,
+      valid_programming_params: programming_params
+    } do
+      Enum.each([mcq_params, programming_params], fn params ->
```
Comment (indocomsoft) [inline test/cadet/assessments/question_test.exs]: Use `for` list comprehension instead

File: lib/cadet/assessments/question.ex [hunk 2]
```diff
@@ -24,10 +25,35 @@ defmodule Cadet.Assessments.Question do
...
     question
     |> cast(params, @required_fields ++ @optional_fields)
     |> cast_embed(:library)
+    |> add_belongs_to_id_from_model(:assessment, params)
     |> validate_required(@required_fields ++ @required_embeds)
+    |> validate_question_content()
+    |> foreign_key_constraint(:assessment_id)
+  end
+
+  defp validate_question_content(changeset) do
```
[the remark this answers is not in the document]
Comment (tuesmiddt) [reply lib/cadet/assessments/question.ex]: HAHA. Dammit you caught me skiving. Will refactor.

File: coveralls.json [hunk 3]
```diff
@@ -9,4 +9,4 @@
     "test/support",
     "test/factories"
   ]
-}
+}
```
[the remark this answers is not in the document]
Comment (tuesmiddt) [reply coveralls.json]: I thought I configured VSCode to automatically add it...

[the remark this answers is not in the document]
Comment (indocomsoft) [reply coveralls.json]: This is why vim is the best. It won't even show the empty last line lol

**Comment by tuesmiddt** [comment]
Made requested changes @indocomsoft

File: coveralls.json [hunk 4]
```diff
@@ -10,3 +10,4 @@
     "test/factories"
   ]
 }
+
```
Comment (indocomsoft) [inline coveralls.json]: Now it's one newline too many :joy:

Just `git checkout master coveralls.json` haha

File: test/cadet/assessments/question_test.exs [hunk 5]
```diff
@@ -1,49 +1,130 @@
...
+      assert({:ok, _} = res, inspect(res, pretty: true))
+    end
+  end
+
+  describe "invalid changesets" do
+    test "missing params", %{
+      valid_mcq_params: mcq_params,
+      valid_programming_params: programming_params
+    } do
+      for params <- [mcq_params, programming_params] do
```
Comment (indocomsoft) [inline test/cadet/assessments/question_test.exs]: Ehh, it should be
```elixir
for params <- [mcq_params, programming_params],
    field <- @required_fields ++ @required_embeds do
  params_missing_field = Map.delete(params, field)
  refute(
    Question.changeset(%Question{}, params_missing_field).valid?,
    inspect(params_missing_field, pretty: true)
  )
end
```

Comment (tuesmiddt) [reply test/cadet/assessments/question_test.exs]: OHHHHHHHHHHHHHHHHHHHHHHHHHH

File: coveralls.json (hunk 4 above)
Comment (tuesmiddt) [reply coveralls.json]: AHHHHHHHHHHH

**Comment by tuesmiddt** [comment]
alright done

**Comment by tuesmiddt** [comment]
Let's wait until #147 is merged. I'll fix the tests to use the new helper.

**Comment by indocomsoft** [comment]
Alright, I just merged from master. Just make sure you `git pull --rebase` before pushing

**Comment by tuesmiddt** [comment]
@indocomsoft Refactored tests to use new ChangesetCase, should be good to go

File: test/cadet/assessments/question_test.exs [hunk 6]
```diff
@@ -3,64 +3,96 @@ defmodule Cadet.Assessments.QuestionTest do
...
+  describe "invalid changesets" do
+    test "missing params", %{
+      valid_mcq_params: mcq_params,
+      valid_programming_params: programming_params
+    } do
+      for params <- [mcq_params, programming_params],
+          field <- @required_fields ++ @required_embeds do
+        params_missing_field = Map.delete(params, field)
+
+        assert_changeset(params_missing_field, :invalid)
```
Comment (indocomsoft) [inline test/cadet/assessments/question_test.exs]: I think should use pipe syntax

File: test/cadet/assessments/question_test.exs [hunk 7]
```diff
@@ -3,64 +3,96 @@ defmodule Cadet.Assessments.QuestionTest do
...
+
+    test "invalid question content", %{
+      valid_mcq_params: mcq_params,
+      valid_programming_params: programming_params
+    } do
+      mcq_params = Map.put(mcq_params, :type, :programming)
+      assert_changeset(mcq_params, :invalid)
+
+      programming_params = Map.put(programming_params, :type, :mcq)
+      assert_changeset(programming_params, :invalid)
```
Comment (indocomsoft) [inline test/cadet/assessments/question_test.exs]: Pipe syntax haha

File: test/cadet/assessments/question_test.exs [hunk 8]
```diff
@@ -3,64 +3,96 @@ defmodule Cadet.Assessments.QuestionTest do
...
+        assert_changeset(params_missing_field, :invalid)
+      end
+    end
+
+    test "invalid question content", %{
+      valid_mcq_params: mcq_params,
+      valid_programming_params: programming_params
+    } do
+      mcq_params = Map.put(mcq_params, :type, :programming)
+      assert_changeset(mcq_params, :invalid)
```
Comment (indocomsoft) [inline test/cadet/assessments/question_test.exs]: Pipe syntax

**Pull request merged**